csharp/src/Ice/ObjectAdapter.cs

```diff
@@ -822,9 +824,7 @@ public async Task SetPublishedEndpointsAsync(IEnumerable<Endpoint> newEndpoints)
                     }
                     else
                     {
-                        // TODO: better fallback!
-                        Protocol = Communicator.DefaultProtocol;
-                        Debug.Assert(Protocol != default);
+                        Protocol = fallbackProtocol;
```

Comment:
I don't think the public API lets the application specify this protocol. Maybe it would be cleaner to have 2 ObjectAdapter ctors, one without a name and a protocol and one with a name but without a protocol?

Reply:
Yes, I can fix in a separate PR.